### DISCUSSION TITLE
findUnique record by relation fields

### DISCUSSION THREAD

**Discussion by alucardu** [prisma/prisma#9745] [Q&A] [answered]
I have 3 models: User, Movie, MovieRating:

```prisma
model Movie {
  id             Int            @id @default(autoincrement())
  original_title String         @unique
  tmdb_id        Int            @unique
  poster_path    String         @unique
  release_date   String
  users          User[]
  movieRating    MovieRating[]
}

model User {
  id            Int             @id @default(autoincrement())
  email         String          @unique
  user_name     String          @unique
  password      String
  movies        Movie[]
  movieRating    MovieRating[]
}

model MovieRating {
  id              Int           @id @default(autoincrement())
  value           Int           @default(0)
  user            User[]        @relation(fields: [userId], references: [id])
  userId          Int
  movie           Movie[]       @relation(fields: [movieId], references: [id])
  movieId         Int
}
```
A mutation to add a rating:
```graphql
mutation {
  addMovieRating(
    userId: 581
    movieId: 611
    value: 3
  ) {
    userId
    movieId
    value
  }
}
```
And a query to return the movie rating:
```ts
returnMovieRating: async (_, args, {req}) => {
  return prisma.movieRating.findUnique({
    where: {
      id: 17,
    },
  });
},
```
![image](https://user-images.githubusercontent.com/779921/137126603-83514c3c-207f-4af7-9238-db9956cbecf5.png)

I'm now returning the movie rating based on its id. But in my application I don't know the id of the rating. I want to return the rating based on the `userId` and the `movieId`. Something like this:

```ts
  returnMovieRating: async (_, args, {req}) => {
    return prisma.movieRating.findUnique({
      where: {
        userId: 581,
        movieId: 611,
      },
    });
  },
```

But this returns an error:

> where: {\n    userId: 581,\n    ~~~~~~\n    movieId: 611\n    ~~~~~~~\n  }\n  ~~~~~~~~~~~~~~\n}\n\nArgument where of type MovieRatingWhereUniqueInput needs exactly one argument, but you provided userId and movieId. Please choose one. 

Any suggestions?

**Comment by ryands17** [comment, marked as answer]
@alucardu 👋 

You can use `findFirst` instead of `findUnique` for this.

**Reply by alucardu** [reply]
Thanks @ryands17 that works. Any reason why this wouldn't work with `findUnique`? Seems like a good use case for it. Return a unique value based on some parameters.

**Reply by ryands17** [reply]
`findUnique` as the name suggests, only works with unique fields of the relation. For non-unique fields or relation fields, we created `findFirst`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
